src/service/user.service.js

```diff
@@ -0,0 +1,146 @@
+import { v4 as uuidv4 } from 'uuid';
+import { User } from '../modules/User.js';
+import { emailService } from './email.service.js';
+import { ApiError } from '../exeptions/api.error.js';
+import { ConsoleLogger } from '../untils/consoleLogger.js';
+
+async function findByUser(email, userName) {
+  const emailUser = await User.findOne({ where: { email } });
+  const nameUser = await User.findOne({ where: { userName } });
+
+  return {
+    emailUser,
+    nameUser,
+  };
+}
+
+function validateEmail(value) {
+  const EMAIL_PATTERN = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!value) {
+    return 'Email is required';
+  }
+
+  if (!EMAIL_PATTERN.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+async function register(email, password, userName) {
+  const activationToken = uuidv4();
+
+  const existUser = await findByUser(email, userName);
+
+  if (existUser) {
+    const errorMessage = {};
+
+    if (existUser.email === email) {
+      errorMessage.email = 'Email already exists';
+    }
+
+    if (existUser.userName === userName) {
```

Comment:
The condition `existUser.userName === userName` is incorrect. It should be `existUser.nameUser` to match the structure returned by `findByUser`.